developer/services/disposablevm-implementation.rst

```diff
@@ -2,73 +2,162 @@
 
-Preparing a savefile is done by ``/usr/lib/qubes/qubes_prepare_saved_domain.sh`` script. It takes two mandatory arguments, appvm name (APPVM) and the savefile name, and optional path to “prerun” script. The script executes the following steps:
+Named disposables are useful for service qubes, as referencing static names is easier when the qube name is mentioned on Qrexec policies (:file:`qubes.UpdatesProxy` target) or as a property of another qube, such as a disposable :term:`net qube` which is referenced by downstream clients in the ``netvm`` property.
 
-1. APPVM is started by ``qvm-start``
+Unnamed disposables have their names in the format :samp:`disp{1234}`, where :samp:`{1234}` is derived from the :py:attr:`dispid <core-admin:qubes.vm.dispvm.DispVM.dispid>` property, a random integer ranging from 0 to 9999 with a fail-safe mechanism to avoid reusing the same value in a short period.
 
-2. xenstore key ``/local/domain/appvm_domain_id/qubes_save_request`` is created
+The system and every qube can have the :py:attr:`default_dispvm <core-admin:qubes.vm.dispvm.DispVM.default_dispvm>` property. If the qube property is set to the default value, it will use the system's property. This property can only have disposable template as value or an empty value. Qubes which have this property set are allowed to request the creation of a disposable from this property. An exception to the rule is the property of disposables, which always default to their disposables templates to avoid data leaks such as using unintended network paths.
 
-3. if prerun script was specified, copy it to ``qubes_save_script`` xenstore key
+There are some Qrexec services that which allows execution to disposables created from the :py:attr:`default_dispvm <core-admin:qubes.vm.dispvm.DispVM.default_dispvm>` property when the destination qube of the Qrexec field uses the :doc:`@dispvm <core-qrexec:qrexec-policy>` tag, most commonly used to open files and URLs, (:file:`qubes.OpenInVM` and :file:`qubes.OpenURL`, respectively).
 
-4. wait for the ``qubes_used_mem`` key to appear
+Preloaded disposables
+---------------------
 
-5. (in APPVM) APPVM boots normally, up to the point in ``/etc/init.d/qubes_core`` script when the presence of ``qubes_save_request`` key is tested. If it exists, then
 
-   1. (in APPVM) if exists, prerun script is retrieved from the respective xenstore key and executed. This preloads filesystem cache with useful applications, so that they will start faster.
+Preloaded disposables are :term:`internal <internal qube>` :term:`unnamed disposables <unnamed disposable>` started in the background and kept hidden from the user when not in use. They are interrupted (paused or suspended, as appropriate) and resumed (transparently) when a disposable qube is requested by the user.
```

Comment:
And similar to an earlier remark - first say why (started in the background, prepared to be ready when user request it), and then how (internal, paused/suspended etc).